revise 'component definition' to 'component declaration' (#351)
oam-dev/spec: 7.application_configuration.md

A component can be deployed into numerous different runtimes. However, at runtime there is a one-to-one correspondence between an _instance of a configuration_ and an _application configuration_.

> Conceptually speaking, an application configuration is managed as part of the _application operator_ role. An application configuration includes one or more component instances, each represented by a component declaration that defines how an instance of a component spec should be deployed. The component declaration has three parts:

Reviewer: Good catch! I think we can make it even more simpler, how about: 
```suggestion
Conceptually speaking, an application configuration is managed as part of the _application operator_ role. An application configuration references one or more component instances, each represented by a section that defines how an instance of a component spec should be deployed. In detail, this section has three parts:
```

Author: Good.Now it looks more clear and no ambiguity.
